### PR TITLE
Remove kie parent 7 from OptaPlanner 8

### PR DESCRIPTION
**Pull request by ge0ffrey**
Changes in this PR:

* Remove kie-parent 7 as the parent from optaplanner-parent. Use jboss-parent directly.
* Upgrade jboss-parent from 31 to 36
* Copy all plugin definitions and other info (except for dependencies) from kie-parent into optaplanner-parent, except for gwt related and other obvious removals (such as felix, pressgang, etc)
* Remove OSGi support
* Copy all dependencies that are actually used by optaplanner from kie-parent to optaplanner-build-parent.
* Upgrade asciidoc plugin from 2.0.0.RC1 to 2.0.0

What hacks did this PR had to do:
* Comment out parent checkstyle config. It's commented out in the pom.xml
* Disable the enforcer rule for plugin versions. It has false negatives.

What is **not in scope** of this PR (=future PR's):
* Clean up plugins
* Clean up dependencies, but "potentially bad dependencies" were marked with a TODO though.
* Upgrade plugin versions
* Upgrade dependency versions
* Switch to kogito-drools instead of drools 7

How reliable are these changes? Might we loose an enforcer rule, code coverage check, sonar check or something?
Good question. I don't know. I've done it pretty diligently, but a lot could be broken. We need to check.

### CODE REVIEW

**Comment by ge0ffrey** [comment]
Githbub actions build is failing like this, which is wierd:

`[ERROR] Failed to execute goal on project optaplanner-persistence-common: Could not resolve dependencies for project org.optaplanner:optaplanner-persistence-common:jar:8.0.0-SNAPSHOT: Could not find artifact org.optaplanner:optaplanner-core:jar:tests:8.0.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/) -> [Help 1]`

File: optaplanner-docs/pom.xml [hunk 1]
```diff
@@ -42,52 +42,6 @@
 
     <pluginManagement>
       <plugins>
-        <plugin>
```
Comment (rsynek) [inline optaplanner-docs/pom.xml:45]: What is the motivation for moving the asciidoc configuration from here to the optaplanner-parent?

File: optaplanner-build-parent/pom.xml [hunk 2]
```diff
@@ -14,15 +14,268 @@
...
+    <version.org.jdom>1.1.3</version.org.jdom>
+    <version.org.jfree.jfreechart>1.5.0</version.org.jfree.jfreechart>
+    <version.org.junit>5.5.2</version.org.junit>
+    <version.org.mockito>1.10.19</version.org.mockito>
+    <version.org.slf4j>1.7.26</version.org.slf4j>
+    <version.org.springframework>5.2.6.RELEASE</version.org.springframework>
+    <version.org.springframework.boot>2.2.6.RELEASE</version.org.springframework.boot>
+  </properties>
+
   <dependencyManagement>
```
[the remark this answers is not in the document]
Comment (ge0ffrey) [reply optaplanner-build-parent/pom.xml:44]: Agreed, out of scope for this PR though. that's in a next step.

File: optaplanner-docs/pom.xml (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-docs/pom.xml:45]: We agreed to move the <configuration> bit back into optaplanner-docs

File: optaplanner-build-parent/pom.xml [hunk 3]
```diff
@@ -14,16 +14,1365 @@
...
+        </plugin>
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>exec-maven-plugin</artifactId>
+          <version>1.5.0</version>
+        </plugin>
+        <plugin>
+          <groupId>org.sonarsource.scanner.maven</groupId>
+          <artifactId>sonar-maven-plugin</artifactId>
+          <version>3.6.0.1398</version>
```
Comment (rsynek) [inline optaplanner-build-parent/pom.xml:796]: version defined in jboss-parent

File: optaplanner-build-parent/pom.xml [hunk 4]
```diff
@@ -14,16 +14,1365 @@
...
+        <!--                <argLine>-Xmx1024m -Dfile.encoding=UTF-8</argLine>-->
+        <!--                <systemPropertyVariables>-->
+        <!--                  <apple.awt.UIElement>true</apple.awt.UIElement>-->
+        <!--                </systemPropertyVariables>-->
+        <!--              </configuration>-->
+        <!--            </plugin>-->
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-failsafe-plugin</artifactId>
+          <version>${version.failsafe.plugin}</version>
```
[the remark this answers is not in the document]
Comment (ge0ffrey) [reply optaplanner-build-parent/pom.xml:688]: +1, same for exec plugin too

**Pull request merged**